src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php

```diff
@@ -34,14 +34,26 @@ class DelegatingLoader extends BaseDelegatingLoader
     /**
      * Constructor.
      *
+     * Ability to pass a LoggerInterface instance as second argument will be removed in 3.0.
+     *
      * @param ControllerNameParser    $parser   A ControllerNameParser instance
-     * @param LoggerInterface         $logger   A LoggerInterface instance
      * @param LoaderResolverInterface $resolver A LoaderResolverInterface instance
      */
-    public function __construct(ControllerNameParser $parser, LoggerInterface $logger = null, LoaderResolverInterface $resolver)
+    public function __construct(ControllerNameParser $parser, $resolver, $r = null)
     {
         $this->parser = $parser;
-        $this->logger = $logger;
+
+        if ($resolver instanceof LoggerInterface) {
+            $this->logger = $resolver;
+            $resolver = $r;
+
+            @trigger_error('Passing a LoggerInterface instance a second argument of the '.__METHOD__.' method is now deprecated and will not be supported anymore in 3.0.', E_USER_DEPRECATED);
+        } else if (func_num_args() >= 3) {
+            throw new \InvalidArgumentException(sprintf(
+                'Expected argument of type "LoggerInterface", "%s" given',
```

Comment:
this is useless. existing legacy code could only inject a LoggerInterface because of the typehint before. And for new code we do not need to validate that since it doesn't matter.